scripts/qemucommand.py

```diff
@@ -166,6 +186,11 @@ def command_line(self):
             cmdline += ['-cpu', 'Haswell']
         if self.overlay:
             cmdline.append(self.overlay)
+
+        # If booting with u-boot is disabled, add kernel command line arguments through qemu -append option
+        if not self.enable_u_boot:
+            cmdline += ["-append", "root=/dev/vda rw highres=off mem=256M console=ttyS0 "
+                                   "ip=one oprofile.timer=1 uvesafb.task_timeout=-1"]
```

Comment:
Why hardcore the `mem` here? I think we can safely rely on the existing mechanism for that. And what is the rest doing? All of `ip=one oprofile.timer=1 uvesafb.task_timeout=-1` looks mysterious to me.

Reply:
**ip=one** is a  mistake, must be **ip=dhcp**! Without this option, the network interface is not started.
This is a reference to kernel command-line arguments 
https://www.kernel.org/doc/html/v4.14/admin-guide/kernel-parameters.html
which for this option redirect to this doc 
https://www.kernel.org/doc/Documentation/filesystems/nfs/nfsroot.txt

**"oprofile.timer"** and **"uvesafb.task_timeout=-1"** are introduced here
https://git.yoctoproject.org/cgit.cgi/poky/tree/meta/conf/machine/include/qemuboot-x86.inc
and actually both these arguments can be removed

**oprofile.timer=1** is related to this tool https://linux.die.net/man/1/oprofile
the man page says:
`OProfile makes use of the hardware performance counters provided on Intel, AMD, and other 
processors, and uses a timer-interrupt based mechanism on CPUs without counters.`
**It can be removed!**

**uvesafb** - is related to the frame buffer driver for the console which is not necessary for our case an also can be removed

**mem** argument, you're right, this value is set by default.